plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -39,6 +61,102 @@ export type EntityRefLinkProps = {
   children?: React.ReactNode;
 } & Omit<LinkProps, 'to'>;
 
+type PeekAheadPopoverProps = {
+  popupState: PopupState;
+  entityRef: CompoundEntityRef;
+};
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '20em',
+    },
+  };
+});
+
+export const PeekAheadPopover = ({
+  popupState,
+  entityRef,
+}: PeekAheadPopoverProps) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+  const [entity, setEntity] = useState<Entity | undefined>();
+
+  const { loading, error } = useAsync(async () => {
+    if (!entity) {
+      if (popupState.isOpen) {
+        const catalogApi = apiHolder.get(catalogApiRef);
+        if (catalogApi) {
+          const retrievedEntity = await catalogApi.getEntityByRef(entityRef);
+          if (!retrievedEntity) {
+            throw new Error(`${entityRef.name} was not found`);
+          }
+          setEntity(retrievedEntity);
+        }
+      }
+    }
+  }, [popupState.isOpen, apiHolder, entityRef]);
+
+  return (
+    <HoverPopover
+      PaperProps={{
+        className: classes.popoverPaper,
+      }}
+      {...bindPopover(popupState)}
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'center',
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+    >
+      <Card>
+        {loading && <Progress />}
+        <CardContent>
+          <Typography color="textSecondary">{entityRef.namespace}</Typography>
+          <Typography variant="h5" component="div">
+            {entityRef.name}
```

Comment:
Note that this ref will not have the right casing. It'll say "component" instead of "Component" for example, and mixed-case names will be wrong too. It should show the real data that you got from the catalog instead. Maybe you want to hold off on rendering the entire card contents until loading finishes